### PR TITLE
fix(fileuploaderitem): use refs for ellipsis, add fowardRef to `Text`

### LINKED ISSUE #16844: [Bug]: FileUploaderItem not working in ShadowDOM
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox

### Package version

1.60.1

### React version

18.3.1

### Description

We use the shadowDOM.
When I update beyond `1.39.0` (`1.39.0` works for us), I have an issue with the `FileUploaderItem` component - this component works fine if not using shadowDOM.
The issue is with line https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/FileUploader/FileUploaderItem.tsx#L111
```
const element = document.querySelector(`[title="${name}"]`);
```
In the shadowDOM, `element` returns as `null `and later the Carbon component tries to do `setIsEllipsisApplied(element.offsetWidth < element.scrollWidth);` - this is where it bombs out with the error: 
```
TypeError: Cannot read properties of null (reading 'offsetWidth')
    at isEllipsisActive (FileUploaderItem.js:72:34)
    at eval (FileUploaderItem.js:77:5)
    at commitHookEffectListMount (react-dom.development.js:23184:26)
    at commitLayoutEffectOnFiber (react-dom.development.js:23302:17)
    at commitLayoutMountEffects_complete (react-dom.development.js:24722:9)
    at commitLayoutEffects_begin (react-dom.development.js:24708:7)
    at commitLayoutEffects_begin (react-dom.development.js:24690:11)
    at commitLayoutEffects (react-dom.development.js:24646:3)
    at commitRootImpl (react-dom.development.js:26857:5)
    at commitRoot (react-dom.development.js:26716:5)
```

### Reproduction/example

https://stackblitz.com/edit/react-ts-mss5gp?file=index.tsx

### Steps to reproduce

Just open the stackblitz link and you will see the error message.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by guidari**
Closes #16844

#### Changelog

**New**

- Added `forwardRef` to Text component.

**Changed**

- Convert the `document.querySelector` to `useRef` in `FileUploaderItem`

#### Testing / Reviewing

- Check if there is no regressions in `FileUploaderItem`. You can check the Drag and Drop examples in storybook.

### CODE REVIEW

**Review by 2nikhiltom — APPROVED** [review]
LGTM  ✅

**Review by emyarod — APPROVED** [review]
looks good to me! maybe need to update title/commit message for merge if there's a CI check for that

File: packages/react/src/components/Text/Text.tsx [hunk 1]
```diff
@@ -6,10 +6,11 @@
  */
 
 import PropTypes from 'prop-types';
-import React, { ReactNode, useContext } from 'react';
+import React, { ReactNode, useContext, useRef } from 'react';
 import { PolymorphicProps } from '../../types/common';
 import { TextDir } from './TextDirection';
 import { TextDirectionContext } from './TextDirectionContext';
+import { useMergeRefs } from '@floating-ui/react';
```
Comment (tay1orjones) [inline packages/react/src/components/Text/Text.tsx:13]: Does this have full equivalency with the internal helper of the same name?

**Review by tay1orjones — APPROVED** [review]
+1 on Andrew's comment, commits/titles not in the proper format will not be picked up by the changelog script and won't be included in the changelog

File: packages/react/src/components/Text/Text.tsx (hunk 1 above)
Comment (tay1orjones) [reply packages/react/src/components/Text/Text.tsx:13]: I opened https://github.com/carbon-design-system/carbon/issues/17234 to cover this

**Pull request merged**